tests/timeseries/_ev_charger_pool/test_ev_charger_pool_control_methods.py

```diff
@@ -0,0 +1,266 @@
+    await microgrid._data_pipeline.initialize(
+        ResamplerConfig(resampling_period=timedelta(seconds=0.1))
+    )
+    streamer = MockComponentDataStreamer(mockgrid.mock_client)
+
+    dp = typing.cast(_DataPipeline, microgrid._data_pipeline._DATA_PIPELINE)
+
+    yield _Mocks(
+        mockgrid,
+        streamer,
+        dp._ev_power_wrapper.status_channel.new_sender(),
+    )
+
+
+class TestEVChargerPoolControl:
+    """Test the EV charger pool control methods."""
+
+    async def _patch_ev_pool_status(
+        self,
+        mocks: _Mocks,
+        mocker: MockerFixture,
+        component_ids: list[int] | None = None,
+    ) -> None:
+        """Patch the EV charger pool status.
+
+        If `component_ids` is not None, the mock will always return `component_ids`.
+        Otherwise, it will return the requested components.
+        """
+        if component_ids:
+            mock = MagicMock(spec=ComponentPoolStatusTracker)
+            mock.get_working_components.return_value = component_ids
+            mocker.patch(
+                "frequenz.sdk.actor.power_distributing._component_managers"
+                "._ev_charger_manager._ev_charger_manager.ComponentPoolStatusTracker",
+                return_value=mock,
+            )
+        else:
+            mock = MagicMock(spec=ComponentPoolStatusTracker)
+            mock.get_working_components.side_effect = set
+            mocker.patch(
+                "frequenz.sdk.actor.power_distributing._component_managers"
+                "._ev_charger_manager._ev_charger_manager.ComponentPoolStatusTracker",
+                return_value=mock,
+            )
+        await mocks.component_status_sender.send(
+            ComponentPoolStatus(working=set(mocks.microgrid.evc_ids), uncertain=set())
+        )
+
+    async def _patch_data_pipeline(self, mocker: MockerFixture) -> None:
+        mocker.patch(
+            "frequenz.sdk.microgrid._data_pipeline._DATA_PIPELINE._ev_power_wrapper"
+            "._pd_wait_for_data_sec",
+            0.1,
+        )
+
+    async def _patch_power_distributing_actor(
+        self,
+        mocker: MockerFixture,
+    ) -> None:
+        dp = typing.cast(_DataPipeline, microgrid._data_pipeline._DATA_PIPELINE)
+        pda = typing.cast(
+            PowerDistributingActor, dp._ev_power_wrapper._power_distributing_actor
+        )
+        cm = typing.cast(
+            EVChargerManager,
+            pda._component_manager,
+        )
+        mocker.patch(
+            "frequenz.sdk.microgrid._data_pipeline._DATA_PIPELINE._ev_power_wrapper"
+            "._power_distributing_actor._component_manager._config",
+            EVDistributionConfig(
+                component_ids=cm._config.component_ids,
+                initial_current=cm._config.initial_current,
+                min_current=cm._config.min_current,
+                increase_power_interval=timedelta(seconds=0.12),
+            ),
+        )
```

Comment:
Also so much need for patching IMHO shows a design issue. Ideally we should test in a way that we can just pass fake arguments to functions directly instead of patching so much global state.

Reply:
True.  To be able to pass this as an argument to `EVChargerManager`, I'd have to change the abstract class for creating `ComponentManager`.  Made an issue: https://github.com/frequenz-floss/frequenz-sdk-python/issues/910